metadb/db_mongodb.go

```diff
@@ -1,14 +1,16 @@
-// +build mongodb
+//
 
 package metadb
 
 import (
+	"os"
+
 	tmdb "github.com/tendermint/tm-db"
 	"github.com/tendermint/tm-db/mongodb"
 )
 
 func mongoDBCreator(name, dir string) (tmdb.DB, error) {
-	return mongodb.NewDB(name, dir, name)
+	return mongodb.NewDB(os.Getenv("MONGO_URL"), name, name)
```

Comment:
So here we make it depend on an environment variable. Why not add a parameter in the function? Is this only for testing purposes?

Reply:
the tm-db api did not designed a way to config the db(url, username ,password, sock) to be created, it will be refactored later